### ISSUE TITLE
Error converting dates with fractional seconds

### ISSUE THREAD

**Issue by LevN0** [Small-Bodies-Node/pds4_tools#6]
User Report:

Came across another small bug in the PDS4 tools, which affects date formats as: 2018-12-12T08:51:16.28972Z (5 decimals) when trying to plot.

The issue is in pds4_tools.reader.data_types.data_type_convert_dates on line 652, where I believe that:

```
for i in range(0, 6):
    format_lengths[max(format_lengths.keys()) + i] = format
```

should be

```
for i in range(0, 6):
    format_lengths[max(format_lengths.keys()) + 1] = format
```

otherwise the key for the lengths of formats with various numbers of decimals is calculated incorrectly - specifically for the above case I get a key error due to the length (25) not being in the format_lengths dictionary.

**Comment by LevN0** [comment]
User is correct in diagnosing cause of the bug. There is another related bug, which allows for extra search of potential date format with fractional seconds due to the delimiter (.) before %f.

**Comment by LevN0** [comment]
Resolved by baefe2463a69a13ba5c4cbc5fb539fa2973d49fe.
